megamek/src/megamek/common/Mounted.java

```diff
@@ -254,13 +255,35 @@ public EquipmentType getType() {
         return (null != type) ? type : (type = EquipmentType.get(typeName));
     }
 
+    public int getModesCount() {
+        return type.getModesCount();
+    }
+
+    protected EquipmentMode getMode(int mode) {
+        return type.getMode(mode);
+    }
+
+    public boolean hasModes() {
+        return type.hasModes();
```

Comment:
I wonder if these three "type" should a) be turned into getType() for consistency?

And also I wonder if for safety we want to check for null in here? Maybe not because it's used in a lot of places and hasn't had a problem before, but just thinking out loud.

Reply:
Agreed. type is doesn't get serialized and getType() will reinitialize it from the equipment lookup name, so it should be accessed through getType() throughout the class.